lib/jsonSchema.ts

```diff
@@ -1,127 +1,159 @@
+const fromJsonSchemaField =
+  (jsonSchema: JSONSchema4) =>
+  (fieldName: string, fieldValue: JSONSchema4): FieldDefinition => {
     if (!isJsonSchemaSupported(fieldValue)) {
-        throw new UnsupportedJsonSchemaError(`Field: ${fieldName} has an unsupported schema`);
+      throw new UnsupportedJsonSchemaError(
+        `Field: ${fieldName} has an unsupported schema`
+      )
     }
-    const optional = !isJsonSchemaRequired(jsonSchema)(fieldName);
+    const optional = !isJsonSchemaRequired(jsonSchema)(fieldName)
 
     switch (fieldValue.type) {
-        case 'string':
-            return fields.createStringField(optional);
-        case 'integer':
-        case 'number':
-            return fields.createIntField(64, optional);
-        case 'boolean':
-            return fields.createBooleanField(optional);
-        case 'array':
-            return fromJsonSchemaArray(fieldValue, optional);
-        case 'object':
-            return fields.createStructField(fromJsonSchema(fieldValue), optional);
-        default:
-            throw new UnsupportedJsonSchemaError(
-                `Unable to convert "${fieldName}" with JSON Schema type "${fieldValue.type}" to a Parquet Schema.`,
-            )
+      case "string":
+        if (fieldValue.format && fieldValue.format == "date-time") {
+          return fields.createTimestampField(optional)
+        }
+        return fields.createStringField(optional)
+      case "integer":
+      case "number":
+        return fields.createIntField(64, optional)
+      case "boolean":
+        return fields.createBooleanField(optional)
+      case "array":
+        return fromJsonSchemaArray(fieldValue, optional)
+      case "object":
+        return fields.createStructField(fromJsonSchema(fieldValue), optional)
+      default:
+        throw new UnsupportedJsonSchemaError(
+          `Unable to convert "${fieldName}" with JSON Schema type "${fieldValue.type}" to a Parquet Schema.`
+        )
     }
-}
+  }
 
 /**
  * Converts supported Json Schemas into Parquet Schema Definitions
  */
 export const fromJsonSchema = (jsonSchema: JSONSchema4): SchemaDefinition => {
-    if (!isJsonSchemaSupported(jsonSchema)) {
-        throw new UnsupportedJsonSchemaError("Unsupported fields found");
-    }
+  if (!isJsonSchemaSupported(jsonSchema)) {
+    throw new UnsupportedJsonSchemaError("Unsupported fields found")
+  }
 
-    const schema: SchemaDefinition = {};
+  const schema: SchemaDefinition = {}
 
-    const fromField = fromJsonSchemaField(jsonSchema)
+  const fromField = fromJsonSchemaField(jsonSchema)
 
-    for (const [fieldName, fieldValue] of Object.entries(
-        jsonSchema.properties || {},
-    )) {
-        schema[fieldName] = fromField(fieldName, fieldValue);
-    }
+  for (const [fieldName, fieldValue] of Object.entries(
+    jsonSchema.properties || {}
+  )) {
+    schema[fieldName] = fromField(fieldName, fieldValue)
+  }
 
-    return schema;
```

Comment:
Also just noting that we will eventually get a linter and formatter, but not today.
https://github.com/LibertyDSNP/parquetjs/issues/34